src/js/renderers/NavigatorLayout.js

```diff
@@ -39,31 +40,14 @@ export const gutenbergNavigatorLayoutTester = rankWith(
 const MemoizedChildComponent = ( ( { component, label, path, contentType } ) => {
     const navigator = useNavigator();
 
-    // Below 2 conditions are hard coded to handle the array renderers
+    // Handle array renderer
     if ( contentType === 'array' ) {
         return (
-            component.data ? (
-                range( 0, component.data.length ).map(( index ) => {
-                    return (
-                        <NavigationButtonAsItem
-                            path={ `${ navigator.location.path }/${ index }` }
-                            aria-label={label}
-                        >
-                            <HStack justify="space-between">
-                                <FlexItem>
-                                    item #{index}
-                                </FlexItem>
-                                <IconWithCurrentColor
-                                    icon={ isRTL() ? chevronLeft : chevronRight}
-                                />
-                            </HStack>
-                        </NavigationButtonAsItem>
-                    )
-                }) 
-            ) : null
+            <ArrayControlRenderer { ...component }/>
```

Comment:
Let `ArrayControlRenderer` handle the rendering, so far it is the list of items and `add new` button